src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -1677,5 +1677,98 @@ public static boolean isXnp(final String reference, final String alternate) {
+     *  exclusively with -1.
+     */
+    public static int[] matchAllelesOnly(final VariantContext variant1, VariantContext variant2) {
+        Utils.nonNull(variant1);
+        Utils.nonNull(variant2);
+
+        // Grab the trivial case:
+        if (variant1.isBiallelic() && variant2.isBiallelic()) {
+            if (variant1.getAlternateAllele(0).equals(variant2.getAlternateAllele(0)) &&
+                    (variant1.getReference().equals(variant2.getReference()))) {
+                return new int[]{0};
+            } else {
+                return new int[]{-1};
+            }
+        }
+
+        // Handle the case where one or both of the input VCs are not biallelic.
+        final int[] result = new int[variant1.getAlternateAlleles().size()];
+
+        // First split (and trim) all variant contexts into biallelics.  We are only going ot be interested in the alleles.
+        final List<VariantContext> splitVariants1 = simpleSplitIntoBiallelics(variant1);
+        final List<VariantContext> splitVariants2 = simpleSplitIntoBiallelics(variant2);
+
+        // Second, match on ref and alt.  If match occurs add it to the output list.
+        for (int i = 0; i < splitVariants1.size(); i++) {
+            result[i] = -1;
+            for (int j = 0; j < splitVariants2.size(); j++) {
+                final VariantContext splitVariant1 = splitVariants1.get(i);
+                final VariantContext splitVariant2 = splitVariants2.get(j);
+                if (splitVariant1.getAlternateAllele(0).equals(splitVariant2.getAlternateAllele(0))
+                        && splitVariant1.getReference().equals(splitVariant2.getReference())) {
+                    result[i] = j;
+                }
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Do not use this method for doing a full split of a variant context into biallelic components.  This method
+     *  ignores (and drops) genotypes and INFO attributes.  It is really meant just for alleles, but works on a
+     *  VariantContext to better leverage existing code in
+     *  {@link GATKVariantContextUtils#trimAlleles(VariantContext, boolean, boolean)}
+     *
+     * This method is trying to be fast.
+     *
+     * @param vc variant context to split into simple biallelics.  Never {@code null}
+     * @return a list of variant contexts.  Each will be biallelic.  Length will be the number of alt alleles in the input vc.
+     * Note that the variant contexts are usually stripped of attributes and genotypes.  Never {@code null}.  Empty list
+     * if variant context has no alt alleles.
+     */
+    private static List<VariantContext> simpleSplitIntoBiallelics(final VariantContext vc) {
+        Utils.nonNull(vc);
+        final List<VariantContext> result = new ArrayList<>();
+
+        for (final Allele allele : vc.getAlternateAlleles()) {
+            result.add(
+                    (vc.isBiallelic()) ? vc :
```

Comment:
Why not just have an if statement for this case and return?  Then you don't have to enter the `for` loop at all.

Reply:
Done